### DISCUSSION TITLE
CloudBytes Help / Discussion Community

### DISCUSSION THREAD

**Discussion by rehanhaider** [CloudBytes-Academy/CloudBytes.dev#50] [Announcements]
Need any help with topics associated with Python🐍 or Cloud ☁☁. Feel free to start a discussion.

**Comment by Kujiranoai** [comment]
### Running a .yml file with a github action in a devcontainer

Hello - I was working through your article on [Automate Deployment of Pelican Website to Github pages](https://cloudbytes.dev/articles/automate-deployment-of-pelican-website-to-github-pages) which uses a vscode devcontainer and everything was going fine until it came to actually running the yaml file, pelican.yml.

I created a .github/workflows/pelican.yml file under the .devcontainer folder and followed the instructions and while the git push command worked OK the instructions in the pelican.yml file never seemed to get triggered. I tried various things such as rerunning the devcontainer from scratch after setting up the file but with no luck.

My github repository with the relevant files is [here:](https://github.com/Kujiranoai/pelicantest) and the [link to pelican.yml](https://github.com/Kujiranoai/pelicantest/blob/main/.devcontainer/.github/workflows/pelican.yml) is here.

I'm not so expert at this so might be making some simple error somewhere but if you had any advice it would be much appreciated, and thanks for your article.

**Reply by rehanhaider** [reply]
@Kujiranoai: You need to keep your `.github` folder in your project root not inside the `.devcontainer` folder.

They are for two different purposes, i.e. the `.devcontainer` is for the local development environment and is needed only for VSCode / Docker on your own PC. 
GitHub uses `.github/workflows` folder to house it's Actions so it needs to be in the root folder of your project. 

Let's me know if this works, or if you need more help.

**Reply by rehanhaider** [reply]
@Kujiranoai for example, take a look at the [CloudBytes.dev source code repository here](https://github.com/CloudBytesDotDev/CloudBytes.dev) and notice `.devcontainer` and `.github` are two different folders. 

And while you're at it, star the repository if this helped 🙂

**Reply by Kujiranoai** [reply]
Thanks very much for replying so quickly.

I changed the location of the .githhub/workflows as you suggested and the blog showed up in the relevant github pages section.  

Unfortunately although the blog posts are there the CSS did not come through (although I can see the css files being produced and in github and appropriate cross references to them in the HTML post file) so that is a new problem to think about - but I think that's enough for me for today...

Best regards

**Reply by rehanhaider** [reply]
Where are you storing your CSS files? I didn't see them on the repository.

**Reply by Kujiranoai** [reply]
Hi - [here under output - theme - css](https://github.com/Kujiranoai/pelicantest/tree/main/output/theme/css).  

The pelican html pages reference theme/cssfilename.css which seems natural in the circumstances - but perhaps there is some problem in referencing the file...?

**Reply by Kujiranoai** [reply]
Hello - I finally got it working thanks to your article and help, thanks very much. It worked after I changed the SITEURL variable to the github pages URL and the relative references to True in the pelican publish config file.  All beginners errors I suspect. Changes to the content are feeding through OK so I'll play around with it more...

Thanks again and regards

**Reply by rehanhaider** [reply]
Congratulations!!

**Reply by rehanhaider** [reply]
@Kujiranoai I happened to look at your repository and will give you one suggestion. 

Keep your theme / CSS outside of the `output` folder. `output` folder is ephemeral and is generated everytime you update the content. The GitHub action is designed in a way where you don't need to upload the output folder at all to GitHub (refer to the CloudBytes repository, it doesn't have an output folder). Instead the action generates it on the fly and then uploads it into the relevant place.

**Comment by jalsco** [comment]
About the pelican minify  : [https://cloudbytes.dev/snippets/pelican-plugin-to-minify-html-css-and-js](url)

For me it fails on first step:  `pip install minify-html`

`ERROR: Could not find a version that satisfies the requirement minify-html (from versions: none)`
`ERROR: No matching distribution found for minify-html`

I am running Debian 10 (buster) and python 3.7.3

**What steps should I take to try to resolve this problem ?** 

I was going to use :  css-html-js-minify:  But read your comment "Broke the website 💔, removed embedded JS, deleted some CSS variables & class definitions." ... so I do not want to risk that

**Reply by rehanhaider** [reply]
This is because older versions of Python is not supported by the package. You can do the following
```bash
pip install minify-html==0.6.10
```

It should word with Python 3.7

**Reply by jalsco** [reply]
Firstly  **THANK YOU !** for replying. I know that it is not your package. I really appreciate that you sent answer.  But.. still not working.

`pip install minify-html==0.6.10`
`ERROR: Could not find a version that satisfies the requirement minify-html==0.6.10 (from versions: none)`
`ERROR: No matching distribution found for minify-html==0.6.10`

**Reply by jalsco** [reply]
On my Debian 10 (diet-pi)  it says that I am up to the latest level of Python that is supported  `Python 3.7.3`

**Reply by rehanhaider** [reply]
OK, try this once. Seems they also changed the package name
```bash
pip install minify_html
```

If this doesn't work, check if you're on any proxy network where PyPI is blocked for downloads.

**Reply by jalsco** [reply]
Thanks again for your help !  Seems still not working... 

![image](https://user-images.githubusercontent.com/7985459/143563563-adbd3fd5-aa4a-4a92-b32a-30fb6da48581.png)

I tried with the `_` and also specifying `==0.6.10`

I am not on any proxy network that I know of .. I did not have any trouble before using `pip install`
But I will check with someone if possible to change the network somehow just to try install.  I doubt it is possible... But I will try

**Reply by rehanhaider** [reply]
Are you able to install any other pip package?

**Reply by jalsco** [reply]
yes , I install pip packages all the time. Never had any problem with that. I checked the network - that server is not network proxy-ed at all. I believe it is set to use opendns  for its DNS. But I never had difficulty with `pip install` starting up.  On the rare occasions that `pip` fails (not counting this `minify_html` )  it has been well into an install, never failed to connect before...

**Reply by rehanhaider** [reply]
I understand, can you try installing something right now please? 

I can install the package so my guess would some issue with the local mirror of PyPI or network issues. 

![minifyhtml](https://user-images.githubusercontent.com/4152163/143566071-ec42809d-8259-4470-8b06-f09a7159567b.png)

**Reply by jalsco** [reply]
I am on `ARM` processor not `AMD` as in your screenshot.   Maybe they have not compiled binaries for `ARM`, I do not know... 

Yes, I just successfully did the `pip install` and `pip uninstall` of the  `css-html-js-minify` so I do know that pip is working

**Reply by rehanhaider** [reply]
Right, that is a possibility. You might want to raise an issue in [minify-html](https://github.com/wilsonzlin/minify-html) repo. I've found them to be typically quite responsive.

**Reply by jalsco** [reply]
thanks . Issue #53 🙏🏻👍🏻

**Comment by w3rm** [comment]
thanks for writing this article -> https://cloudbytes.dev/snippets/upgrade-python-to-latest-version-on-ubuntu-linux, it helped me a lot!

**Reply by rehanhaider** [reply]
@w3rm Appreciate your kind words.

**Reply by jalsco** [reply]
Sorry for asking a simple question... Just about formatting in RST in that article :

![image](https://user-images.githubusercontent.com/7985459/143570698-10f9de72-4c55-43c5-be72-f8f1378fa59f.png)

How did you make the space on the line before the `apt list` ?  When I do it the `apt` is right up against the left edge of that box.

**Reply by rehanhaider** [reply]
@jalsco that is done using CSS. you need `padding :0.5em;` or something like that

**Comment by Hoklifter** [comment]
I dont have my linux anymore :( , any way of getting it back?

After following

 https://cloudbytes.dev/snippets/upgrade-python-to-latest-version-on-ubuntu-linux

My pc's ambient doesnt work anymore. Someone can help me? Im new to linux 


![IMG_20221107_093214.jpg](https://user-images.githubusercontent.com/114830688/200311241-9969804b-5dcf-40b2-91a4-dce4d10ad459.jpg)

My linux is a mint 20.3.
